app/controllers/sign_up_controller.rb

```diff
@@ -8,6 +8,19 @@ def create
     end
   end
 
+  def login
+    @user = User.find_by(email: params[:email])
+    if @user&.valid_password?(params[:password])
```

Comment:
Where is the method `valid_password?` defined? Devise? Devise will be removed.

Reply:
yes it is a device, when the device is removed, I will change the method to authenticate, because it does not work with the device

Reply:
I will add has_secure_password to model user and method authenticate with this